lib/libthr/thread/thr_sig.c

```diff
@@ -294,6 +305,9 @@ handle_signal(struct sigaction *actp, int sig, siginfo_t *info, ucontext_t *ucp)
 	__sys_sigprocmask(SIG_SETMASK, &actp->sa_mask, NULL);
 
 	sigfunc = actp->sa_sigaction;
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(RTLD_SANDBOX)
+	assert(cheri_getperm(sigfunc) & CHERI_PERM_EXECUTIVE);
```

Comment:
Is this a check of a property enforced elsewhere or a security relevant check? If the latter, you'll want to use abort() directly as assert() can be compiled out when compiled with NDEBUG defined.